tests/queries/test_db_returning.py

```diff
@@ -26,6 +26,11 @@ def test_insert_returning_non_integer(self):
         self.assertTrue(obj.created)
         self.assertIsInstance(obj.created, datetime.datetime)
 
+    def test_insert_returning_non_integer_from_literal_value(self):
+        obj = NonIntegerPKReturningModel.objects.create(pk="2025-01-01")
+        self.assertTrue(obj.created)
+        self.assertIsInstance(obj.created, datetime.datetime)
```

Comment:
I exploded this out into a test case so that if we yank it later for being "dubious" we can. Should I add a little comment that this usage is a little dubious? 🤔

Reply:
I was accustomed to seeing `UUIDField`s used like this with string representations on queryset-level operations (although not in a db_returning context) so maybe this is fine as is without a comment.

Reply:
I mean, it's always dubious to make ORM calls with the unexpected type (in the sense that it's not the type `to_python` and `full_clean` would provide if called) but since the lax typing enforcement has likely ossified in code bases it's more of an artifact of some database backends accepting the string representation of dates, uuid, and others as input.

Reply:
Yeah, the pattern:

```
def view(request, book_id: str):
    obj = get_object_or_404(Book, id=book_id)
```

…is super-common and relies on the database casting string to ID. I agree we can probably never remove this behaviour.

I blogged about one case where type conversion caused an outage, and some investigation across databases: https://adamj.eu/tech/2020/03/06/sql-implicit-type-conversion/